pkg/authorization/apibinding_authorizer.go

```diff
@@ -51,32 +52,22 @@ const (
 // exported resources workspace. If it is not allowed we will return NoDecision, if allowed we
 // will call the delegate authorizer.
 func NewAPIBindingAccessAuthorizer(kubeInformers kubernetesinformers.SharedInformerFactory, kcpInformers kcpinformers.SharedInformerFactory, delegate authorizer.Authorizer) (authorizer.Authorizer, error) {
-	if _, found := kcpInformers.Apis().V1alpha1().APIBindings().Informer().GetIndexer().GetIndexers()[byWorkspaceIndex]; !found {
-		err := kcpInformers.Apis().V1alpha1().APIBindings().Informer().AddIndexers(
-			cache.Indexers{
-				byWorkspaceIndex: func(obj interface{}) ([]string, error) {
-					return []string{logicalcluster.From(obj.(metav1.Object)).String()}, nil
-				},
-			},
-		)
-		if err != nil {
-			// nothing we can do here. But this should also never happen. We check for existence before.
-			return nil, fmt.Errorf("failed to add indexer for APIBindings: %w", err)
-		}
-	}
-	if _, found := kcpInformers.Apis().V1alpha1().APIExports().Informer().GetIndexer().GetIndexers()[byWorkspaceIndex]; !found {
-		err := kcpInformers.Apis().V1alpha1().APIExports().Informer().AddIndexers(
-			cache.Indexers{
-				byWorkspaceIndex: func(obj interface{}) ([]string, error) {
-					return []string{logicalcluster.From(obj.(metav1.Object)).String()}, nil
-				},
-			},
-		)
-		if err != nil {
-			// nothing we can do here. But this should also never happen. We check for existence before.
-			return nil, fmt.Errorf("failed to add indexer for APIExports: %w", err)
-		}
-	}
+	apiBindingIndexer := kcpInformers.Apis().V1alpha1().APIBindings().Informer().GetIndexer()
+	apiExportIndexer := kcpInformers.Apis().V1alpha1().APIExports().Informer().GetIndexer()
+
+	indexers.AddIfNotPresentOrDie(
```

Comment:
Since you're changing this... every shared informer generally has `indexers.ByLogicalCluster` and `indexers.ByLogicalClusterAndNamespace` preconfigured.

Reply:
oh i missed that one, I believe I can then even change that to:
```
indexers.AddIfNotPresentOrDie(apiBindingIndexer, indexers.ClusterScoped())
indexers.AddIfNotPresentOrDie(apiExportIndexer, indexers.ClusterScoped())
```